### PR TITLE
Update to risc0 1.2.1-rc.0

### PR DESCRIPTION
**Pull request by flaub**

### CODE REVIEW

File: guests/reth-ethereum/Cargo.toml [hunk 1]
```diff
@@ -30,7 +30,8 @@ lazy_static = "1.5.0"
 reth-chainspec = { git = "https://github.com/risc0/reth", branch = "p1.1.0_zstd", default-features = false, features = ["std"] }
 
 [patch.crates-io]
+c-kzg = { git = "https://github.com/risc0/c-kzg-4844.git", branch = "p1.0.3" }
 crypto-bigint = { git = "https://github.com/risc0/RustCrypto-crypto-bigint", tag = "v0.5.5-risczero.0" }
-k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", tag = "k256/v0.13.3-risczero.0" }
+k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", rev = "c2750428558944954af5fe147b01a67711a65062" }
```
Comment (austinabell) [inline guests/reth-ethereum/Cargo.toml:35]:
```suggestion
k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", tag = "k256/v0.13.3-risczero.1" }
```

File: guests/reth-optimism/Cargo.toml [hunk 2]
```diff
@@ -30,7 +30,8 @@ lazy_static = "1.5.0"
 reth-optimism-chainspec = { git = "https://github.com/risc0/reth", branch = "p1.1.0_zstd", default-features = false, features = ["std"] }
 
 [patch.crates-io]
+c-kzg = { git = "https://github.com/risc0/c-kzg-4844.git", branch = "p1.0.3" }
 crypto-bigint = { git = "https://github.com/risc0/RustCrypto-crypto-bigint", tag = "v0.5.5-risczero.0" }
-k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", tag = "k256/v0.13.3-risczero.0" }
+k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", rev = "c2750428558944954af5fe147b01a67711a65062" }
```
Comment (austinabell) [inline guests/reth-optimism/Cargo.toml:35]:
```suggestion
k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", tag = "k256/v0.13.3-risczero.1" }
```

File: testing/ef-tests/testguest/Cargo.toml [hunk 3]
```diff
@@ -6,11 +6,12 @@ edition = "2021"
...
 [dependencies]
-risc0-zkvm = { version = "1.1.3", default-features = false, features = ['std'] }
+risc0-zkvm = { version = "1.2.1-rc.0", default-features = false, features = [ "std", "unstable" ] }
 zeth-lib = { path = "../../../lib", default-features = false }
 
 [patch.crates-io]
 # use optimized risc0 circuit
 crypto-bigint = { git = "https://github.com/risc0/RustCrypto-crypto-bigint", tag = "v0.5.5-risczero.0" }
-k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", tag = "k256/v0.13.3-risczero.0" }
+k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", rev = "c2750428558944954af5fe147b01a67711a65062" }
```
Comment (austinabell) [inline testing/ef-tests/testguest/Cargo.toml:15]:
```suggestion
k256 = { git = "https://github.com/risc0/RustCrypto-elliptic-curves", tag = "k256/v0.13.3-risczero.1" }
```

**Review by austinabell — COMMENTED** [review]
The k256 patch can be updated to the released tag if you want.

Also, seems cached RPC data was committed.

Happy to resolve both of these myself if you're okay with me pushing to your branch

**Comment by flaub** [comment]
Sounds good. Pushing cached blocks was intentional.

**Comment by austinabell** [comment]
> Sounds good. Pushing cached blocks was intentional.

Works, just updated the patch tag in that case :)

**Review by Wollac — APPROVED** [review]

**Pull request merged**